### PR TITLE
Bookstore: Use Redux in React components.

### PR DESCRIPTION
**Pull request by jlcarrascof**
# Implementing Redux in React Components of the Bookstore Project

This pull request implements the usage of **Redux** in the React components of the Bookstore project. The Redux store is connected to the app to provide access to state variables (accessed through `useSelector`) and state modifiers (a combination of a `reducer` and `useDispatch`). The integration of Redux in this project allows for more intricate state management and enhances code maintainability.

## Changes Made

The following changes have been implemented:

📦 Wrapped the `<App>` component with the `react-redux` `<Provider>` component.

📋 Passed the configured store into the `<Provider>`.

🏁 Created an `initialState` variable for `booksSlice`.

📚 Displayed the books retrieved from the *slice* in a reusable component.

🚀 Dispatched actions using `useDispatch`.

➕ Added a `<Button>` component that includes an event handler to add a book to the books array (with attributes `id`, `title`, and `author`). In my code, due to my approach, this component is actually the `AddBookForm` component.

🗑️ Included a `<Button>` component that incorporates an event handler to remove a book from the books array (by its `id`). In my code, due to my approach, this component is actually the `Book` component.

🔧 Refactored other components to utilize Redux and reflect the implemented changes.

### CODE REVIEW

File: src/components/BooksList.js [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
-      {data.forEach((book) => {
-        <Book title={book.title} author={book.author} />;
-      })}
-    </>
+    <div>
+      <h2>Books list</h2>
+      <ul>
+        {data.value.map((book) => (
+          <Book
+            key={book.item_id}
```
Comment (BregornOriginal) [inline src/components/BooksList.js:14]: - [ ] Kindly fix this error in the browser. As far I can see, it seems an error from your `key` value which is equal to `item1`. You could use `uuid` to create a random `id` 💯 

image

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
Hi @jlcarrascof  

Good job so far!

There are some issues that you still need to work on to go to the next project but you are almost there!

giphy (5)

## Highlights

✅ Linters are passing
✅ Good commit history
✅ Good Pull Request description
✅ Excellent work implementing redux 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BooksList.js (hunk 1 above)
Comment (jlcarrascof) [reply src/components/BooksList.js:14]: Ohhh boy, yes ... 🙈🙈 I'll work on it Mr @BregornOriginal Julio 🙂🫡👍

Comment (jlcarrascof) [reply src/components/BooksList.js:14]: By error I left in this code:

Book.propTypes = {
  book: PropTypes.shape({
    **item_id: PropTypes.string.isRequired,**
    title: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
  }).isRequired,
};

item_id: as number .... **Now is fixed and working. Adding a new record, with an unique ID and generated by UUID.**


ERROR-FIXED

**Review by abdulhamiid — APPROVED** [review]
# :green_circle:  Approved! :partying_face:
Hi @jlcarrascof 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Descriptions are descriptive
- ✔️ Use store, actions, and reducers in React.
- ✔️ Connect React and Redux

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**